### PR TITLE
Integrate async refactor work - (Async Refactor Part 3)

### PR DESCRIPTION
**Pull request by dhaval24**
Merge inorder before merging:
1. #816 
2. #817 

This PR integrates 816, 817 and build should pass.

### CODE REVIEW

File: web/src/main/java/com/microsoft/applicationinsights/web/extensibility/modules/WebSessionTrackingTelemetryModule.java [hunk 1]
```diff
@@ -50,18 +48,16 @@ public void initialize(TelemetryConfiguration configuration) {
 
     /**
      * Begin request processing.
-     *
-     * @param req The request to process
+     *  @param req The request to process
```
[the remark this answers is not in the document]
Comment (dhaval24) [reply web/src/main/java/com/microsoft/applicationinsights/web/extensibility/modules/WebSessionTrackingTelemetryModule.java]: This is ill effect of IDE.... I will fix

File: core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java [hunk 2]
```diff
@@ -119,6 +120,7 @@ public void testDockerContextNotAccessedIfPollerNotCompleted() {
         verify(contextPollerMock, times(0)).getDockerContext();
     }
 
+    @Ignore("Tests run flaky on Mac OS")
```
[the remark this answers is not in the document]
Comment (dhaval24) [reply core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: No. I moved all my development to mac and discovered that these tests fail once in a while. So disabling for now until they are robust.

**Comment by dhaval24** [comment]
@reyang I have addressed all the comments. Let me know if there is anything else.

File: core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (reyang) [reply core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: This should not affect CI right? Better restore them as currently it only affects your dev environment.

[the remark this answers is not in the document]
Comment (dhaval24) [reply core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: Done

File: core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java [hunk 3]
```diff
@@ -29,6 +29,7 @@
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
```
Comment (reyang) [inline core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: clean up this.

Comment (dhaval24) [reply core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: done

**Pull request merged**